### ISSUE TITLE
I can't add to a model with many to many relationship using datastore. It works for models without relationships

### ISSUE THREAD

**Issue by CloudInnovator** [aws-amplify/amplify-js#13061]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

DataStore

### Amplify Version

v6

### Amplify Categories

api

### Backend

Amplify CLI

### Environment information

<details>

```
# Put output below this line
System:
    OS: Linux 6.5 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
    Memory: 4.00 GB / 15.47 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.8.0 - ~/.nvm/versions/node/v20.8.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v20.8.0/bin/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.8.0/bin/npm
    pnpm: 8.12.1 - /usr/local/bin/pnpm
  Browsers:
    Chrome: 121.0.6167.139
  npmPackages:
    @aws-amplify/ui-react: ^6.1.2 => 6.1.3 
    @aws-amplify/ui-react-internal:  undefined ()
    @babel/plugin-proposal-class-properties: ^7.16.12 => 7.18.6 
    @babel/plugin-proposal-private-methods: ^7.16.12 => 7.18.6 
    @babel/plugin-proposal-private-property-in-object: ^7.21.11 => 7.21.11 (7.21.0-placeholder-for-preset-env.2)
    @babel/plugin-transform-private-property-in-object: ^7.23.4 => 7.23.4 
    @chakra-ui/icons: ^2.1.1 => 2.1.1 
    @chakra-ui/react: ^2.8.2 => 2.8.2 
    @emotion/react: ^11.11.3 => 11.11.3 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @fortawesome/fontawesome-free: ^6.5.1 => 6.5.1 
    @fortawesome/fontawesome-svg-core: ^6.5.1 => 6.5.1 
    @fortawesome/free-solid-svg-icons: ^6.5.1 => 6.5.1 
    @fortawesome/react-fontawesome: ^0.2.0 => 0.2.0 
    @reduxjs/toolkit: ^2.1.0 => 2.2.0 
    @reduxjs/toolkit-query:  1.0.0 
    @reduxjs/toolkit-query-react:  1.0.0 
    @reduxjs/toolkit-react:  1.0.0 
    @syncfusion/ej2: ^19.4.48 => 19.4.56 
    @syncfusion/ej2-react-calendars: ^19.4.48 => 19.4.56 
    @syncfusion/ej2-react-charts: ^19.4.50 => 19.4.54 
    @syncfusion/ej2-react-dropdowns: ^19.4.52 => 19.4.56 
    @syncfusion/ej2-react-grids: ^19.4.50 => 19.4.56 
    @syncfusion/ej2-react-inputs: ^19.4.52 => 19.4.52 
    @syncfusion/ej2-react-kanban: ^19.4.48 => 19.4.52 
    @syncfusion/ej2-react-popups: ^19.4.52 => 19.4.53 
    @syncfusion/ej2-react-richtexteditor: ^19.4.50 => 19.4.56 
    @syncfusion/ej2-react-schedule: ^19.4.50 => 19.4.55 
    @testing-library/jest-dom: ^5.17.0 => 5.17.0 
    @testing-library/react: ^13.4.0 => 13.4.0 
    @testing-library/user-event: ^13.5.0 => 13.5.0 
    antd: ^5.13.3 => 5.14.1 
    aws-amplify: ^6.0.13 => 6.0.16 
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    aws-sdk: ^2.1554.0 => 2.1556.0 
    chart.js: ^4.4.1 => 4.4.1 
    chart.js-auto:  undefined ()
    chart.js-helpers:  undefined ()
    express: ^4.18.2 => 4.18.2 
    formik: ^2.4.5 => 2.4.5 
    framer-motion: ^11.0.3 => 11.0.3 
    graphql: ^16.8.1 => 16.8.1 (15.8.0)
    jmespath: ^0.16.0 => 0.16.0 
    papaparse: ^5.4.1 => 5.4.1 
    platform: ^1.3.6 => 1.3.6 
    ra-data-graphql: ^4.16.8 => 4.16.10 
    react: ^18.2.0 => 18.2.0 
    react-admin: ^4.16.5 => 4.16.10 
    react-chartjs-2: ^5.2.0 => 5.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    react-icons: ^4.3.1 => 4.12.0 
    react-modal: ^3.16.1 => 3.16.1 
    react-redux: ^9.1.0 => 9.1.0 
    react-responsive: ^9.0.2 => 9.0.2 
    react-router-dom: ^6.21.3 => 6.22.0 
    react-scripts: 5.0.1 => 5.0.1 
    react-select: ^5.8.0 => 5.8.0 
    react-table: ^7.8.0 => 7.8.0 
    react-toastify: ^10.0.4 => 10.0.4 
    styled-components: ^6.1.8 => 6.1.8 
    styled-components/native:  undefined ()
    ua-parser: ^0.3.5 => 0.3.5 
    ua-parser-js: ^1.0.37 => 1.0.37 
    web-vitals: ^2.1.4 => 2.1.4 
    yup: ^1.3.3 => 1.3.3 
  npmGlobalPackages:
    @aws-amplify/cli: 12.10.1
    corepack: 0.20.0
    npm: 10.1.0
    serve: 14.2.1
    yarn: 1.22.21

```

</details>

### Describe the bug

I  am not able to save and to the data store with the following schema:
```

type Blog @model {
  id: ID!
  name: String!
  posts: [Post] @hasMany
}

type Post @model {
  id: ID! 
  title: String!
  blog: Blog @belongsTo
  comments: [Comment] @hasMany
}

type Comment @model {
  id: ID!
  post: Post @belongsTo
  content: String!
}
type Admin @model {
  id: ID! 
  given_name: String!
  last_name: String!
  email: AWSEmail!
  ip_address: AWSIPAddress!
  device_type: String!
  device_browser: String!
  device_model: String!
  device_make: String!
  phone_number: String!
  birthdate: AWSDate
  entity: String!
  device_vendor: String!
  browser_version: String!
  os_name: String!
  os_version: String!
  cpu_architecture: String!
  os_engine_name: String!
  os_engine_version: String!
  ua: String!
  user: String!
  groups: [String!]! 
  roles: [Role] @manyToMany(relationName: "AdminRole")
  entities: [Entity] @manyToMany(relationName: "AdminEntity")
}

type Role @model {
  id: ID!
  ip_address: AWSIPAddress!
  device_type: String!
  device_browser: String!
  device_model: String!
  device_make: String!
  role: String!
  device_vendor: String!
  browser_version: String!
  os_name: String!
  os_version: String!
  cpu_architecture: String!
  os_engine_name: String!
  os_engine_version: String!
  ua: String!
  user: String!
  groups: [String!]!
  admins: [Admin] @manyToMany(relationName: "AdminRole")
  masses: [Mass] @manyToMany(relationName: "MassRole")
}

type Entity @model {
  id: ID!
  given_name: String
  last_name: String
  email: AWSEmail
  ip_address: AWSIPAddress!
  device_type: String!
  device_browser: String!
  device_model: String!
  device_make: String!
  phone_number: String
  birthdate: AWSDate
  entity: String!
  device_vendor: String!
  browser_version: String!
  os_name: String!
  os_version: String!
  cpu_architecture: String!
  os_engine_name: String!
  os_engine_version: String!
  ua: String!
  user: String!
  groups: [String!]!  
  admins: [Admin] @manyToMany(relationName: "AdminEntity")
  masses: [Mass] @manyToMany(relationName: "MassEntity")
}

type Mass @model {
  id: ID! 
  given_name: String!
  last_name: String!
  email: AWSEmail!
  ip_address: AWSIPAddress!
  device_type: String!
  device_browser: String!
  device_model: String!
  device_make: String!
  phone_number: String
  birthdate: AWSDate
  entity: String!
  device_vendor: String!
  browser_version: String!
  os_name: String!
  os_version: String!
  cpu_architecture: String!
  os_engine_name: String!
  os_engine_version: String!
  ua: String!
  user: String!
  groups: [String!]! 
  roles: [Role] @manyToMany(relationName: "MassRole")
  entities: [Entity] @manyToMany(relationName: "MassEntity")  
}
```
I have checked that my models are being formed properly. I save like so to make sure that save operation is being completed, saved is successful, or failed, but I don't receive any logs.

```
console.log(modelFields);

DataStore.observe(Entity, Predicates.ALL).subscribe(msg => {
  console.log(msg.model, msg.opType, msg.element);
});

DataStore.save(new Entity(modelFields))
  .then(() => {
    console.log("Save Entity successful");
  })
  .catch((error) => {
    console.error("Save Entity failed:", error);
  })
  .finally(() => {
    console.log("Save operation finished");
  });
```

### Expected behavior

Successfully saving to models and then saving to the underhood created amplify join tables/models. The join tables exist. I can see them from the  data content pane in Amplify studio.

### Reproduction steps

1. create schema
2. save dummy data
3. Notice that no error is displayed nor operation is completed or failed. 

### Code Snippet

```javascript
// Put your code below this line.
 <Grid
      as="form"
      rowGap="15px"
      columnGap="15px"
      padding="20px"
      onSubmit={async (event) => {
        event.preventDefault();
        setIsSubmitting(true);
        //addEnumValue("DEAN", "Entity");

        try {


          let modelFields = {
            given_name,
            last_name,
            email,
            ip_address: await logIpAddress(),
            device_type: await getDeviceType(),
            device_browser: await getBrowserName(),
            device_model: await getDeviceModel(),
            device_make: await getDeviceMake(),
            phone_number,
            birthdate,
            entity,
            device_vendor: await getDeviceVendor(),
            browser_version: await getBrowserVersion(),
            os_name: await getOsName(),
            os_version: await getOsVersion(),
            cpu_architecture: await getCPUArchitecture(),
            os_engine_name: await getBrowserEngineName(),
            os_engine_version: await getBrowserVersion(),
            ua: await getUA(),
            user: await fetchedUser.username,
            groups: await groupsArray,
            admins: [],
            masses: []
          };

          const validationResponses = await Promise.all(
            Object.keys(validations).reduce((promises, fieldName) => {
              if (Array.isArray(modelFields[fieldName])) {
                promises.push(
                  ...modelFields[fieldName].map((item) =>
                    runValidationTasks(fieldName, item)
                  )
                );
                return promises;
              }
              promises.push(
                runValidationTasks(fieldName, modelFields[fieldName])
              );
              return promises;
            }, [])
          );
          if (validationResponses.some((r) => r.hasError)) {
            return;
          }
          if (onSubmit) {
            modelFields = onSubmit(modelFields);
          }
          try {
            Object.entries(modelFields).forEach(([key, value]) => {
              if (typeof value === "string" && value === "") {
                modelFields[key] = null;
              }
            });
            console.log("modelFields", modelFields);

            DataStore.observe(Entity, Predicates.ALL).subscribe(msg => {
              console.log(msg.model, msg.opType, msg.element);
            });
            DataStore.save(new Entity(modelFields))
              .then(() => {
                console.log("Save Entity successful");
              })
              .catch((error) => {
                console.error("Save Entity failed:", error);
              })
              .finally(() => {
                console.log("Save operation finished");
              });

            if (onSuccess) {
              onSuccess(modelFields);
            }
            if (clearOnSuccess) {
              resetStateValues();
            }
          } catch (err) {
            if (onError) {
              onError(modelFields, err.message);
            }

            console.log("Error creating Entity", err);
          }
          setIsSubmitting(false);

        } catch (error) {
          setIsSubmitting(false);

          console.log("Error setIsSubmitting Entity", error);


        }
      }}


      {...getOverrideProps(overrides, "EntityCreateForm")}
      {...rest}
    >
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

const awsmobile = {
    "aws_project_region": "eu-west-1",
    "aws_cloud_logic_custom": [
        {
            "name": "AdminQueries",
            "endpoint": "",
            "region": "eu-west-1"
        }
    ],
    "aws_appsync_graphqlEndpoint": "",
    "aws_appsync_region": "eu-west-1",
    "aws_appsync_authenticationType": "API_KEY",
    "aws_appsync_apiKey": "",
    "aws_cognito_identity_pool_id": "eu-west-1:",
    "aws_cognito_region": "eu-west-1",
    "aws_user_pools_id": "eu-west-1",
    "aws_user_pools_web_client_id": "",
    "oauth": {},
    "aws_cognito_username_attributes": [
        "EMAIL"
    ],
    "aws_cognito_social_providers": [],
    "aws_cognito_signup_attributes": [
        "BIRTHDATE",
        "EMAIL",
        "FAMILY_NAME",
        "GIVEN_NAME",
        "PHONE_NUMBER"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [
        "SMS"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": []
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ],
    "Analytics": {
        "AWSPinpoint": {
            "appId": "",
            "region": "eu-west-1"
        }
    }
};

export default awsmobile;

**Comment by CloudInnovator** [comment]
Thanks for getting back to me. I started with ```await```. I only converted to a chained ```then``` version to check if the promise was being resolved or if it was ever completed.  The good news is that both methods worked today. In hindsight, I suspect that my models were in an inconsistent state that day due to alterations after adding data on the fly. I probably needed to rebuild them with ```amplify rebuild api```
